dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -274,6 +287,20 @@ private void validateTrackedEntityAttributeFilters(
     }
   }
 
+  private void validatePagination(PageParams pageParams) throws BadRequestException {
+    if (pageParams == null) {
+      return;
+    }
+
```

Comment:
https://github.com/dhis2/dhis2-core/blob/44fa24febae14f6a20a5d7fcb49e3737d7586e56/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/RequestParamsValidator.java#L574-L586 should actually be done by the `PageParams` constructor 😅 it was ok before when our services were only used by our controllers.

Reply:
I mean we can move the logic of `validatePaginationBounds` into the `PageParams` constructor. Before `PageParams` were validated in the web and only created by the web layer. Now that our service API is used by different modules we need to make sure the `PageParams` that is passed in is well formed. So `PageParams` should have the `validatePaginationBounds` validation. I am not talking about the system limits here 👌🏻 .